.github/workflows/frontend_tests.yml

```diff
@@ -54,24 +54,20 @@ jobs:
           python-version: '3.8.15'
           architecture: 'x64'
       - uses: ./.github/actions/merge
-      - name: Cache node modules and third_party/static
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: |
-            /home/runner/work/oppia/yarn_cache
-            /home/runner/work/oppia/oppia/third_party/static
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock', 'dependencies.json') }}
-          restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
-      - uses: ./.github/actions/install-oppia-dependencies
       - name: Suppress ENOSPC error from chokidar file watcher. See https://stackoverflow.com/a/32600959.
         run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+      # Deleting ubuntu files for getting more space (refer: https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930)
+      - name: Deleting ubuntu files
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+      - name: Initialize Containers
+        run: |
+          sudo make build
+          sudo make run-devserver
+          make stop
```

Comment:
Why do we need to run the server for the fornetend tests?

Reply:
the run-devserver copies the node_modules to local which is required for the FE tests

Reply:
Are we running fronted tests in host or in docker-conntainer? I was expecting it to run in the container!

In case it's running in the container, why do we need to copy the node modules? Also, why do we need to start all the servers (backend/frontend/redis/firebase etc.) just to copy node_modules?

Reply:
@Shivkant-Chauhan You haven't addressed this comment yet.